src/create-select.tsx

```diff
@@ -6,35 +6,32 @@ import {
   on,
 } from "solid-js";
 
-type Option = any;
+type Value<V> = V | V[];
 
-type SingleValue = any;
 
-type Value = SingleValue | SingleValue[];
-
-interface CreateSelectProps {
-  options: Option[] | ((inputValue: string) => Option[]);
-  initialValue?: Value;
+interface CreateSelectProps<O, V> {
+  options: O[] | ((inputValue: string) => O[]);
+  initialValue?: Value<V>;
   multiple?: boolean;
   disabled?: boolean;
-  optionToValue?: (option: Option) => SingleValue;
-  isOptionDisabled?: (option: Option) => boolean;
-  onChange?: (value: Value) => void;
+  optionToValue?: (option: O) => V;
+  isOptionDisabled?: (option: O) => boolean;
+  onChange?: (value: Value<V> | null) => void;
   onInput?: (inputValue: string) => void;
 }
 
-const createSelect = (props: CreateSelectProps) => {
+function createSelect<O, V>(props: CreateSelectProps<O, V>) {
   const config = mergeProps(
     {
       multiple: false,
       disabled: false,
-      optionToValue: (option: Option): SingleValue => option,
-      isOptionDisabled: (option: Option) => false,
+      optionToValue: (option: O): V => option as unknown as V,
```

Comment:
I'm relatively unhappy with this cast/default behavior... but better to describe the existing behavior than just to ignore the error. If I were a better dev I'd know how to make `optionToValue` non-optional if O doesn't extend V.